src/chttpd/src/chttpd.erl

```diff
@@ -431,6 +431,34 @@ handle_req_after_auth(HandlerKey, HttpReq) ->
             {HttpReq, catch_error(HttpReq, ErrorType, Error, Stack)}
     end.
 
+nonce(MochiReq) ->
+    case MochiReq:get_header_value("X-Couch-Request-ID") of
+        undefined ->
+            new_nonce();
+        Value ->
+            case re:run(Value, nonce_regex(), [{capture, none}]) of
+                match ->
+                    Value;
+                nomatch ->
+                    new_nonce()
+            end
+    end.
+
+new_nonce() ->
+    couch_util:to_hex(crypto:strong_rand_bytes(5)).
+
+nonce_regex() ->
+    Key = chttpd_request_id_regex,
+    Regex = config:get("chttpd", "request_id_regex", "^[a-zA-Z0-9-_]{1,64}+$"),
```

Comment:
Wonder how dynamic this pattern needs to be `[a-zA-Z0-9-_]{1,64}+` seems good enough for a general pattern. Maybe just hard-code it and make the start_link put it in the persistent_term?

Reply:
I can't imagine anyone changing it but I didn't want to hardcode it. good idea to pre-set it in start_link.